Sources/Common/Store/SdkConfig.swift

```diff
@@ -17,11 +17,58 @@ public struct SdkConfig {
     // Used to create new instance of SdkConfig when the SDK is initialized.
     // Then, each property of the SdkConfig object can be modified by the user.
     public enum Factory {
-        public static func create(region: Region) -> SdkConfig {
-            SdkConfig(trackingApiUrl: region.productionTrackingUrl)
+        public static func create(region: Region, params: [String: Any] = [:]) -> SdkConfig {
+            // Each SDK config option should be able to be set from `param` map.
+            // If one isn't provided, use the default value instead.
+
+            // If a parameter takes more logic to calculate, perform the logic up here.
+            var logLevel = CioLogLevel.error
+            if let logLevelStringValue = params[Keys.logLevel.rawValue] as? String, let paramLogLevel =
+                CioLogLevel.getLogLevel(for: logLevelStringValue) {
+                logLevel = paramLogLevel
+            }
+
+            // Construct object with all required parameters. Each config option should be provided from `params` or a default value.
+            // Define default values here in constructor instead of in struct properties. This is by design so in the future if we add
+            // a new SDK config option to the struct, we get a compiler error here in the constructor reminding us that we need to
+            // add a way for `params` to override the SDK config option.
+            var config = SdkConfig(
+                trackingApiUrl: (params[Keys.trackingApiUrl.rawValue] as? String) ?? region.productionTrackingUrl,
+                autoTrackPushEvents: (params[Keys.autoTrackPushEvents.rawValue] as? Bool) ?? true,
+                backgroundQueueMinNumberOfTasks: (params[Keys.backgroundQueueMinNumberOfTasks.rawValue] as? Int) ?? 10, backgroundQueueSecondsDelay: params[Keys.backgroundQueueSecondsDelay.rawValue] as? Seconds ?? 30,
+                backgroundQueueExpiredSeconds: params[Keys.backgroundQueueExpiredSeconds.rawValue] as? Seconds ?? Seconds.secondsFromDays(3),
```

Comment:
Something I want to point out, Shahroz, is that `Seconds` data type in a typealias we have setup to be a Double. 

I am not sure if the SDK Wrappers will be able to convert nicely to a Double? Maybe we need to instead have the SDK wrappers provide to us milliseconds in an Int data type that we convert into a Double in native SDK? 

I am thinking the conversion between SDK wrappers and native might not work well with Double data types?